lib/internal/url.js

```diff
@@ -1431,8 +1431,8 @@ function encodePathChars(filepath) {
 }
 
 function pathToFileURL(filepath) {
-  const outURL = new URL('file://');
   if (isWindows && StringPrototypeStartsWith(filepath, '\\\\')) {
```

Comment:
Can StringPrototypeStartsWith be replaced with `filepath.length > 1 && filepath[0] === '\\' && filepath[1] === '\\'`?

Reply:
Yes, but compared to rest of the functionality, the performance impact would be negligible. For example, if we changed the order of this expression to `StringPrototypeStartsWith(filepath, '\\\\') && isWindows`, to benchmark the impact on macOS (because I don't own Windows machine), and take this as a baseline, and apply the changes you've suggested, this would be the impact:

The results show `**` and `*` confidence in accuracy.

```
                                                                                                                 confidence improvement accuracy (*)   (**)  (***)
url/whatwg-url-to-and-from-path.js n=100000 method='fileURLToPath' input='file:///dev/null?key=param&bool'               *      0.97 %       ±0.88% ±1.17% ±1.52%
url/whatwg-url-to-and-from-path.js n=100000 method='fileURLToPath' input='file:///dev/null?key=param&bool#hash'         **      1.22 %       ±0.78% ±1.04% ±1.36%
url/whatwg-url-to-and-from-path.js n=100000 method='fileURLToPath' input='file:///dev/null'                              *      1.73 %       ±1.48% ±1.97% ±2.58%
url/whatwg-url-to-and-from-path.js n=100000 method='pathToFileURL' input='file:///dev/null?key=param&bool'              **      1.26 %       ±0.83% ±1.11% ±1.44%
url/whatwg-url-to-and-from-path.js n=100000 method='pathToFileURL' input='file:///dev/null?key=param&bool#hash'          *      0.81 %       ±0.81% ±1.07% ±1.40%
url/whatwg-url-to-and-from-path.js n=100000 method='pathToFileURL' input='file:///dev/null'                              *      1.24 %       ±0.96% ±1.27% ±1.66%

Be aware that when doing many comparisons the risk of a false-positive result increases.
In this case, there are 6 comparisons, you can thus expect the following amount of false-positive results:
  0.30 false positives, when considering a   5% risk acceptance (*, **, ***),
  0.06 false positives, when considering a   1% risk acceptance (**, ***),
  0.01 false positives, when considering a 0.1% risk acceptance (***)
```

Reply:
But it would have an impact ;)

Reply:
Statistically speaking I’m not `***` confident it will.

Reply:
But I’m happy to approve it a different PR, where we replace more than 1 startsWith.